### PR TITLE
Add getTestWatcher method to TestWorkflowRule

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

## What was changed
Addition of getTestWatcher method to TestWorkflowRule 
## Why?
Currently if you use testworkflowrule you have to create test watcher rule as such for example:

```java
  @Rule
  public TestWorkflowRule testWorkflowRule = ...

  @Rule
  public TestWatcher watchman =
      new TestWatcher() {
        @Override
        protected void failed(Throwable e, Description description) {
          if (testWorkflowRule.getTestEnvironment() != null) {
            System.err.println(testWorkflowRule.getTestEnvironment().getDiagnostics());
            testWorkflowRule.getTestEnvironment().shutdown();
          }
        }
      };
```

This simplifies it to:

```
  @Rule
  public TestWorkflowRule testWorkflowRule = ...

  @Rule
  public TestWatcher watchman = testWorkflowRule.getTestWatcher();
```         

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java [hunk 1]
```diff
@@ -354,4 +354,27 @@ public History getWorkflowExecutionHistory(WorkflowExecution execution) {
...
+   * Returns the default test watcher impl that can be used is your tests. It simplifies the test watcher
+   * creation to just:
+   * <pre><code>
+   *   @Rule
+   *   public TestWatcher watchman = testWorkflowRule.getTestWatcher();
+   *   </code></pre>
+   *
+   * @return default test watcher impl
+   */
+  public TestWatcher getTestWatcher() {
```
Comment (Spikhalskiy) [inline temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java:368]: This method should be called `newTestWatcher` or `supplyTestWatcher`, it's not a getter.

File: temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java [hunk 2]
```diff
@@ -354,4 +354,27 @@ public History getWorkflowExecutionHistory(WorkflowExecution execution) {
...
+    return
+      new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+          if (getTestEnvironment() != null) {
+            System.err.println(getTestEnvironment().getDiagnostics());
+            getTestEnvironment().shutdown();
```
Comment (Spikhalskiy) [inline temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java:375]: Why do we need this? This code is located inside TestWorkflowRule class that takes care of shutting down the environment here, in shutdown call:
```
  @Override
  public Statement apply(Statement base, Description description) {
    taskQueue = init(description);
    Statement testWorkflowStatement =
        new Statement() {
          @Override
          public void evaluate() throws Throwable {
            start();
            base.evaluate();
            shutdown();
          }
        };
    return watchman.apply(globalTimeout.apply(testWorkflowStatement, description), description);
  }
```
So, the environment shutdown is taken care of already by the rule, why do we need also the watcher to handle it?

File: temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java [hunk 3]
```diff
@@ -354,4 +354,27 @@ public History getWorkflowExecutionHistory(WorkflowExecution execution) {
...
+   *
+   * @return default test watcher impl
+   */
+  public TestWatcher getTestWatcher() {
+    return
+      new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+          if (getTestEnvironment() != null) {
+            System.err.println(getTestEnvironment().getDiagnostics());
```
Comment (Spikhalskiy) [inline temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java:374]: Please use logger, there are almost 0 valid cases when non-trivial app should be performing input and output directly to System.err or System.out.

Comment (tsurdilo) [reply temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java:374]: This is ran in tests and from what i see logger statements are swallowed. There is already a System.err in this class on like 87 ;)

File: temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java (hunk 2 above)
Comment (tsurdilo) [reply temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java:375]: This is a separate rule so from what i see needs to be there. On exception the test rule might not shut down.

File: temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java (hunk 1 above)
Comment (tsurdilo) [reply temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java:368]: ok i dont see the big deal but sure. I like that method name.

**Comment by tsurdilo** [comment]
TestWorkflowRule has internal TestWatcher and I think this change is not needed after all :) Will close thanks for review

**Pull request closed without merging**